### PR TITLE
#444  Render siblings on 404 product page

### LINKED ISSUE #444: catalog.py:57-60: Render 404, that is recommending...
The puzzle `273-9b2b9d3f` from #273 has to be resolved:

https://github.com/fidals/shopelectro/blob/36a9dd50f8a4e0685485d57121cf47019dd5646a/shopelectro/views/catalog.py#L57-L60

The puzzle was created by Artemiy on 24-Jul-18. 

Estimate: 0 minutes,     

If you have any technical questions, don't ask me, submit new tickets instead. The task will be "done" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by duker33**
Closes #444

### CODE REVIEW

File: shopelectro/views/catalog.py [hunk 1]
```diff
@@ -77,6 +84,26 @@ def get_context_data(self, **kwargs):
             'group_tags_pairs': group_tags_pairs
         }
 
+    def render_siblings_on_404(
+        self, request, **url_kwargs
+    ) -> typing.Union[http.Http404, None]:
+        """Try to render removed product's siblings on it's 404 page."""
+        product_inactive_qs = models.Product.objects.filter(
```
Comment (artemiy312) [inline shopelectro/views/catalog.py]: We can reduce the code size here
```python
inactive_product = models.Product.objects.filter(...).first()
if inactive_product:
    related_products = ...
```

File: shopelectro/views/catalog.py [hunk 2]
```diff
@@ -77,6 +84,26 @@ def get_context_data(self, **kwargs):
...
+            **{self.slug_field: url_kwargs['product_vendor_code']},
+            category__isnull=False,
+            page__is_active=False
+        )
+        if product_inactive_qs:
+            product = product_inactive_qs.first()
+            related_products = models.Product.objects.filter(
+                category=product.category,
+                page__is_active=True
+            )[10:]
```
Comment (artemiy312) [inline shopelectro/views/catalog.py]: >related_products = models.Product.objects.filter(..)**[10:]**

maybe, did you mean here first ten products instead?
`related_products = models.Product.objects.filter(..)[:10]`

File: shopelectro/views/catalog.py (hunk 1 above)
Comment (duker33) [reply shopelectro/views/catalog.py]: Done.

File: shopelectro/views/catalog.py (hunk 2 above)
Comment (duker33) [reply shopelectro/views/catalog.py]: Done.

**Pull request merged**